core/trino-main/src/main/java/io/trino/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -976,7 +986,7 @@ protected Type visitFunctionCall(FunctionCall node, StackableAstVisitorContext<C
                 ResolvedWindow window = getResolvedWindow.apply(node);
                 checkState(window != null, "no resolved window for: " + node);
 
-                analyzeWindow(window, context, (Node) node.getWindow().get());
+                analyzeWindow(window, new StackableAstVisitorContext<>(context.getContext().skipTrackingSourceColumns()), (Node) node.getWindow().get());
```

Comment:
This is not only about `WINDOW`. It also would need to be handled in https://trino.io/docs/current/functions/aggregate.html#filtering-during-aggregation (where filter is skipped) and `CASE` or `IF` (where condition is skipped). Also there might be other like this in future.

After a second though, I think we should just put all `WINDOW` arguments as source columns for output expression. This makes reasoning easier, code simpler. It will also make it more straightforward for `CASE` and `IF`.